packages/util-package-finder/lib/js/index.js

```diff
@@ -182,37 +136,67 @@ const setStatus = json => {
 };
 
 /**
- * Construct the search URI
+ * Get the interpolated search URI
  * @private
- * @param {String} scope NPM scope to search under
- * @param {Boolean} d show deprecated packages
- * @return {Promise<Array>}
+ * @param {String} subject freetext string to search for
+ * @return {String}
+ */
+const getPackagesSearchURI = subject => {
+	const limit = 250;
+	return `https://registry.npmjs.com/-/v1/search?text=${subject}&size=${limit}`;
+};
+
+/**
+ * Validate default function options
+ * @private
+ * @param {Object} opts search options
+ * @return {Promise<Object>}
  */
-const getURI = (scope, d) => {
-	const deprecated = (d) ? '' : '%20not:deprecated';
-	return `${npmsEndpoint}search?q=scope%3A${scope}${deprecated}&size=250`;
+const validateOptions = opts => {
+	return new Promise((resolve, reject) => {
+		if (!Array.isArray(opts.filters)) {
+			reject(new Error(`Filters parameter must be of type \`array\`, found \`${typeof opts.filters}\``));
+		}
+
+		if (typeof opts.versions !== 'boolean') {
+			reject(new Error(`Versions parameter must be of type \`boolean\`, found \`${typeof opts.versions}\``));
+		}
+
+		resolve(opts);
+	});
+};
+
+/**
+ * Removes any scope prefix
+ * @private
+ * @param {String} scope search scope
+ * @param {String} scope search scope with any scope prefix removed
+ */
+const normaliseScopePrefix = scope => {
+	if (scope.startsWith('@')) {
+		scope = scope.substr(1);
+	}
+	return scope;
 };
 
 /**
- * Get all available packages
+ * Get all non-deprecated packages for a scope, with additional filtering
  * @param {Object} defaults
  * @return {Promise<Array>}
  */
 module.exports = ({
 	scope = 'springernature',
 	filters = [],
-	versions = false,
-	deprecated = false
+	versions = false
 } = {}) => (
 	validateOptions({
-		scope: scope,
+		scope: normaliseScopePrefix(scope),
 		filters: filters,
-		versions: versions,
-		deprecated: deprecated
+		versions: versions
 	})
-		.then(opts => fetch(getURI(opts.scope, deprecated)))
+		.then(opts => fetch(getPackagesSearchURI(opts.scope)))
 		.then(response => response.json())
-		.then(json => filterResults(json, getOptions({scope: scope, filters: filters})))
+		.then(json => filterResults(json, {scope: normaliseScopePrefix(scope), filters: filters}))
```

Comment:
Hasn't this already been done on line 193?
So this just needs to be
```
{scope: scope, filters: filters}
```
or possibly [just this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015)
```
{scope, filters}
```

Reply:
OK, I get why not 😄